concordium-consensus/src/Concordium/MultiVersion.hs

```diff
@@ -939,10 +945,11 @@ getCatchUpRequest = do
 --    of the current genesis index.
 -- 2. Determine the current genesis index before deserializing to ensure
 --    that the correct format is used.
-receiveTransaction :: ByteString -> MVR gsconf finconf UpdateResult
+receiveTransaction :: forall gsconf finconf. ByteString -> MVR gsconf finconf UpdateResult
 receiveTransaction transactionBS = do
     now <- utcTimeToTransactionTime <$> currentTime
-    case runGet (getExactVersionedBlockItem now) transactionBS of
+    SomeProtocolVersion spv <- currentProtocolVersion
+    case runGet (getExactVersionedBlockItem spv now) transactionBS of
```

Comment:
This is not actually a problem, because a transaction cannot deserialize to two different `BlockItem`s in different protocol versions, and `receiveTransaction` must gracefully handle `BlockItem`s that may not be valid in the current protocol version. Documentation added.